localstack/services/kinesis/kinesis_starter.py

```diff
@@ -22,28 +29,42 @@ def start_kinesis(
     :param update_listener: an update listener instance for server proxy
     :param asynchronous: currently unused but required by localstack.services.plugins.Service.start().
     TODO: either make use of this param or refactor Service.start() to not pass it.
+    :param account_id: account ID to use for this instance of Kinesis-Mock
     :returns: A running Kinesis server instance
     """
     global _server
-    if not _server:
-        _server = kinesis_mock_server.create_kinesis_mock_server(persist_path=persist_path)
 
-    _server.start()
-    log_startup_message("Kinesis")
-    port = port or config.service_port("kinesis")
+    if account_id is None:
+        account_id = get_aws_account_id()
+
+    if account_id not in _server:
+        # To support multi-accounts we use separate instance of Kinesis-Mock per account
+        # See https://github.com/etspaceman/kinesis-mock/issues/377
+        if not _server.get(account_id):
+            _server[account_id] = kinesis_mock_server.create_kinesis_mock_server(
+                account_id=account_id, persist_path=persist_path
+            )
+
+        _server[account_id].start()
+        log_startup_message("Kinesis")
+        port = port or config.service_port("kinesis")
 
-    return _server
+        check_kinesis(account_id=account_id)
```

Comment:
This function now seems to be vulnerable to race-conditions. Multiple parallel requests could trigger the creation of a new server simultaneously. While this was already the case, I guess it will become more pressing now since we are creating new instances for each account ID. It might be better to introduce some locking here?